File: string/1461_check_if_a_string_contains_all_binary_codes_of_size_k/1461_check_if_a_string_contains_all_binary_codes_of_size_k.go
package main

import (
	"fmt"
	"math"
)

func hasAllCodes(s string, k int) bool {
	m := map[string]bool{}
	for i := 0; i < len(s)-k+1; i++ {
		m[s[i:i+k]] = true
	}
	return len(m) == int(math.Pow(2, float64(k)))
}

func main() {
	s := "00110110"
	k := 2
	res := hasAllCodes(s, k)
	fmt.Println(res)
}

File: string/1461_check_if_a_string_contains_all_binary_codes_of_size_k/1461_check_if_a_string_contains_all_binary_codes_of_size_k.py
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        m = set()
        for i in range(len(s)-k+1):
            m.add(s[i:i+k])
        return len(m) == (1 << k)


s = "00110110"
k = 2
res = Solution().hasAllCodes(s, k)
print(res)